packages/x-components/src/components/modals/base-modal.vue

```diff
@@ -108,9 +108,11 @@
      * @internal
      */
     protected disableScroll(): void {
-      this.previousBodyOverflow = document.body.style.overflow;
-      this.previousHTMLOverflow = document.documentElement.style.overflow;
-      document.body.style.overflow = document.documentElement.style.overflow = 'hidden';
+      const scrollingElement = document.scrollingElement as HTMLElement;
+      this.previousScrollingElementOverflow = scrollingElement.style.overflow;
+      this.previousScrollingElementScrollbarGutter = scrollingElement.style.scrollbarGutter;
+      scrollingElement.style.overflow = 'hidden';
+      scrollingElement.style.scrollbarGutter = 'stable';
```

Comment:
I didn't know about this property. I love the idea of avoiding the layout shift that happens due to removing the scrollbar, but the issue I see now is that the `html` scrollbar is always visible overlapping the modal.. I changed here the styles to reflect the different elements to try to show you this. The white bar is the `html` scrollbar.

![image](https://user-images.githubusercontent.com/68222542/193523604-b2f8d540-ce75-4d3d-9b8d-ed9563a4011a.png)